### PR TITLE
Copy paths from the [patch] section into the dummy-src

### PR DESCRIPTION
**Pull request by basvandijk**
This allows users to override the source of crates using:

```
[patch.crates-io]
bar = { path = 'my/local/bar' }
```

### CODE REVIEW

File: default.nix [hunk 1]
```diff
@@ -80,6 +80,12 @@ with rec
               )
             );
 
+          patchedSources =
+            lib.optionals (builtins.hasAttr "patch" toplevelCargotoml)
+              (map (po: src + "/" + po.path)
+                (lib.collect (as: lib.isAttrs as && builtins.hasAttr "path" as)
```
Comment (nmattia) [inline default.nix]: Do you know what else than `crates.io` can be patched?  Is it safe to just assume that we'll "vendor" all patches that have a `path`?

Comment (basvandijk) [reply default.nix]: I think it's safe.

File: default.nix [hunk 2]
```diff
@@ -80,6 +80,12 @@ with rec
               )
             );
 
+          patchedSources =
+            lib.optionals (builtins.hasAttr "patch" toplevelCargotoml)
+              (map (po: src + "/" + po.path)
```
Comment (nmattia) [inline default.nix]: note: would be nice to crash if the path is absolute

**Comment by nmattia** [comment]
Does this work with non-virtual manifests? can you add some tests ? (ideally virtual + non-virtual)

**Pull request merged**